api/rpc/server.go

```diff
@@ -23,11 +30,19 @@ type Server struct {
 
 func NewServer(address, port string) *Server {
 	rpc := jsonrpc.NewServer()
+	authHandler := &auth.Handler{
+		Verify: func(ctx context.Context, token string) ([]auth.Permission, error) {
+			// TODO(distractedm1nd/renaynay): implement auth
+			log.Warn("auth not implemented, token: ", token)
+			return DefaultPerms, nil
```

Comment:
This probably has been discussed in API planning somewhere, so might be a silly question.
Looks like `read` == `public` right now. Is it possible that we might need to have  private `read` that allows reads only for a set of users?

Reply:
It's worth considering this idea @distractedm1nd @renaynay

Reply:
this is what I meant by something between read and admin, i am not opposed. seems like the best option.

admin is too high, read too low, write is semantically incorrect

Reply:
do you propose the following levels then?: 

- public
- read (read + public)
- write (read, write, public)
- admin (all perms)

Reply:
I would purpose to keep read/write as access **type** and role as access **level**. So private write should not grant private read. I think the permission that does both is `readwrite`.

For our reduced case it will be:
 -  `public`
 -  private  level
    - `read`
    - `write`
    - `readwrite`
 -  admin level

Reply:
public is the new default, and protected reads are behind `read`

@walldiss users can sign tokens with arbitrary permission sets, there is no actual hierarchy in this PR. If you sign a token with only the `write` perm you cannot use `read` methods. Discussion of whether requesting `write` perms should add `read` by default are in a PR on top of this